apps/cowswap-frontend/src/modules/permit/utils/handlePermit.ts

```diff
@@ -15,13 +17,13 @@ import { HandlePermitParams } from '../types'
 export async function handlePermit(params: HandlePermitParams): Promise<AppDataInfo> {
   const { permitInfo, inputToken, account, appData, generatePermitHook } = params
 
-  if (permitInfo && 'address' in inputToken) {
+  if (isSupportedPermitInfo(permitInfo) && 'address' in inputToken) {
```

Comment:
would it make sense to declare `isSupportedPermitInfo` as `export function isSupportedPermitInfo(p: PermitInfo | undefined): p is PermitInfo` so you don't need the explicit casting?

Reply:
YES!
I tried to do this but didn't know how, thank you!